### DISCUSSION TITLE
Raspotify startup in an infinte loop... Help :-/

### DISCUSSION THREAD

**Discussion by mjhtek** [dtcooper/raspotify#541] [Q&A] [answered]
Hmm... Right. It's been a while since I did anything on Linux / Pi. I've dusted off an old Pi 2B+ though, as I was taken by the idea of running it as Spotify Connect device. 

This is my hardware info:
```
pi@raspberrypi:~ $ cat /proc/cpuinfo
processor       : 0
model name      : ARMv6-compatible processor rev 7 (v6l)
BogoMIPS        : 697.95
Features        : half thumb fastmult vfp edsp java tls
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xb76
CPU revision    : 7

Hardware        : BCM2835
Revision        : 0010
Serial          : 00000000957998d9
Model           : Raspberry Pi Model B Plus Rev 1.2
```
And 
```
pi@raspberrypi:~ $ dpkg --print-architecture
armhf
```

My set up steps are below, but the raspotify service never starts correctly and is in an infinite retry loop until I stop it. Would appreciate any pointers for things to check. 

Briefly I have the latest build of Bullseye Raspberry OS and to get it setup 
have run:
```
sudo apt update
sudo apt upgrade
sudo apt install -y apt-transport-https curl
curl -sSL https://dtcooper.github.io/raspotify/key.asc | sudo apt-key add -v – echo 'deb https://dtcooper.github.io/raspotify raspotify main' | sudo tee /etc/apt/sources.list.d/raspotify.list
sudo apt update
sudo apt install raspotify
```

This is the service status output:

```
pi@raspberrypi:~ $ service raspotify status
● raspotify.service - Raspotify (Spotify Connect Client)
     Loaded: loaded (/lib/systemd/system/raspotify.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) (Result: signal) since Sun 2022-05-01 22:28:44 BST; 9s ago
       Docs: https://github.com/dtcooper/raspotify
             https://github.com/librespot-org/librespot
             https://github.com/dtcooper/raspotify/wiki
             https://github.com/librespot-org/librespot/wiki/Options
    Process: 4536 ExecStart=/usr/bin/librespot (code=killed, signal=ILL)
   Main PID: 4536 (code=killed, signal=ILL)
        CPU: 833ms
```

`signal=ILL` - is that an illegal instruction?? And is there anything I can do about it? 

This is one of the restarts in the log, accessed using `journalctl -u raspotify`:

```
Apr 30 21:48:07 raspberrypi systemd[1]: raspotify.service: Scheduled restart job, restart counter is at 94.
Apr 30 21:48:07 raspberrypi systemd[1]: Stopped Raspotify (Spotify Connect Client).
Apr 30 21:48:07 raspberrypi systemd[1]: Started Raspotify (Spotify Connect Client).
Apr 30 21:48:08 raspberrypi systemd[1]: raspotify.service: Main process exited, code=killed, status=4/ILL
Apr 30 21:48:08 raspberrypi systemd[1]: raspotify.service: Failed with result 'signal'.
```

I also added raspotify to the `sources.list` as the only thread I could find anywhere near to the same as my problem is [this one](https://discourse.osmc.tv/t/librespot-illegal-instruction-on-4-9-113-45-osmc-on-vero-4k/90028/16), so I ran through the 4 steps there, which seemed to download a lot of updates, but didn't fix my problem...

Anyway, when I run `sudo apt update`, I see the raspotify there at Hit3 following (with a few warnings which I don't think are going to be related to the problem I'm experiencing)

```
pi@raspberrypi:~ $ sudo apt update
Hit:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease
Hit:2 http://archive.raspberrypi.org/debian bullseye InRelease
Hit:3 https://dtcooper.github.io/raspotify raspotify InRelease
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
All packages are up to date.
W: Target Packages (main/binary-armhf/Packages) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Translations (main/i18n/Translation-en_GB) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Packages (main/binary-armhf/Packages) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Translations (main/i18n/Translation-en_GB) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:2 and /etc/apt/sources.list.d/raspotify.list:1

```

I now feel like I'm at a dead end unless I can get some help on this.

**Comment by JasonLG1979** [comment, marked as answer]
A Raspberry Pi Model B Plus Rev 1.2 is a Pi 1 not a Pi 2.

Raspotify no longer supports ARMv6 that is clearly stated in the [readme](https://github.com/dtcooper/raspotify#support-for-armv6-pi-v1-and-pi-zero-v1x-has-been-dropped).

To elaborate. ARMv6 processors are not supported by upstream Debian armhf(true armhf = ARMv7). The Raspberry Pi Foundation sprinkles in their own sauce and repackages everything to make a distro that is ARMv6 compatible.

Raspotify is built with upstream Debian libs not Raspberry Pi OS libs. Mainly because ARMv6 is a dead platform that is no longer even supported by ARM and the toolchain is really not maintained.

**Reply by JasonLG1979** [reply]
The upcoming rewrite of librespot will not even compile for ARMv6 because ARMv6 is a dead platform.

**Reply by mjhtek** [reply]
OK cheers for the info @JasonLG1979. Appreciate it - will have saved me some time digging further into fundamentally unfixable issues. No point flogging a dead horse! I shall upgrade my Pi! I did check the readme, but genuinely thought my Pi was Pi 2. Stupid oversight! In the meantime though, I'd just like to try out raspotify, so may try the old unsupported release v[0.31.8.1](https://github.com/dtcooper/raspotify/releases/tag/0.31.8.1) as long as all the old dependencies are still available I guess I ought to be able to get this running...

**Reply by d3m0-sm** [reply]
> OK cheers for the info @JasonLG1979. Appreciate it - will have saved me some time digging further into fundamentally unfixable issues. No point flogging a dead horse! I shall upgrade my Pi! I did check the readme, but genuinely thought my Pi was Pi 2. Stupid oversight! In the meantime though, I'd just like to try out raspotify, so may try the old unsupported release v[0.31.8.1](https://github.com/dtcooper/raspotify/releases/tag/0.31.8.1) as long as all the old dependencies are still available I guess I ought to be able to get this running...

Im struggeling with an raspberr pi Zero w. It is also on the armv6 architecture, but i use version 0.31.8.1. Can you tell me, wich dependencies you use?

The crash report says that it couldn't start cause of the dependencies

**Reply by JasonLG1979** [reply]
As clearly stated in the [readme](https://github.com/dtcooper/raspotify#support-for-armv6-pi-v1-and-pi-zero-v1x-has-been-dropped) and [wiki page](https://github.com/dtcooper/raspotify/wiki/Raspotify-on-Pi-v1's-and-Pi-Zero-v1.x) Raspotify on ARMv6 is completely supported. To quote the wiki:

**Unsupported means unsupported. You're on your own. Issues, questions and discussions will be closed and locked.**
